profiles/common/modules/features/splash_screen/splash_screen.module

```diff
@@ -70,6 +70,13 @@ function splash_screen_theme($existing, $type, $theme, $path) {
  * Implements hook_preprocess().
  */
 function splash_screen_preprocess_splash(&$variables) {
+  // Set our custom title.
+  $splash_screen_title = variable_get('splash_screen_title_value', variable_get('site_name'));
```

Comment:
I'm sorry, I asked you to simplify this line, but I found problem: If the title field is filled, and after is emptied, the variable exist and the value is "" (empty string).
So you first version was better (empty value is checked)
```
  // Cache our custom title.
  // Set our custom title.
  $custom_splash_screen_title = variable_get('splash_screen_title_value', FALSE);
  $splash_screen_title = variable_get('splash_screen_title_value', variable_get('site_name'));
  // Fallback to the site name if it is not configured.
  $splash_screen_title = $custom_splash_screen_title ? $custom_splash_screen_title : variable_get('site_name');
```

Really sorry for that lost time !